app.rb

```diff
@@ -0,0 +1,274 @@
+          break
+        end
+      end
+    end
+    boolean
+  end
+
+  def my_any?(argument = nil)
+    boolean = false
+    if argument.nil?
+      if block_given?
+        my_each do |element|
+          if yield(element)
+            boolean = true
+            break
+          end
+        end
+      else
+        my_each do |element|
+          unless element.nil? || element == false
+            boolean = true
+            break
+          end
+        end
+      end
+    elsif argument.class == Regexp
+      my_each do |element|
+        if argument.match?(element)
+          boolean = true
+          break
+        end
+      end
+    elsif argument.class == Integer || argument.class == Float
+      my_each do |element|
+        if element == argument
+          boolean = true
+          break
+        end
+      end
+    elsif argument.class == String
+      my_each do |element|
+        if element == argument
+          boolean = true
+          break
+        end
+      end
+    else
+      my_each do |element|
+        if element.class == argument
+          boolean = true
+          break
+        end
+      end
+    end
+    boolean
+  end
+
+  def my_none?(argument = nil)
+    boolean = true
+    if argument.nil?
+      if block_given?
+        my_each do |element|
+          if yield(element)
+            boolean = false
+            break
+          end
+        end
+      else
+        my_each do |element|
+          unless element.nil? || element == false
+            boolean = false
+            break
+          end
+        end
+      end
+    elsif argument.class == Regexp
+      my_each do |element|
+        if argument.match?(element)
+          boolean = false
+          break
+        end
+      end
+    elsif argument.class == Integer || argument.class == Float
+      my_each do |element|
+        if element == argument
+          boolean = false
+          break
+        end
+      end
+    else
+      my_each do |element|
+        if element.class == argument
+          boolean = false
+          break
+        end
+      end
+    end
+    boolean
+  end
+
+  def my_count(argument = nil)
+    counter = 0
+    if block_given?
+      if argument.nil?
+        my_each do |element|
+          yield(element) && counter += 1
+        end
+      else
+        my_each do |element|
+          argument == element && counter += 1
+        end
+      end
+    elsif argument.nil?
+      counter = to_a.length
+    else
+      my_each do |element|
+        argument == element && counter += 1
+      end
+    end
+    counter
+  end
+
+  def my_map(proc1 = nil)
+    if block_given?
+      new_arr = []
+      my_each do |element|
+        new_arr << if proc1.nil?
+                     yield(element)
+                   else
+                     proc1.call(element)
+                   end
+      end
+      new_arr
+    else
+      to_enum :my_map
+    end
+  end
+
+  def my_inject(initial = nil, argument = nil)
```

Comment:
## my_inject:

this method throws the following error:  
`undefined local variable or method `length' for 5..50:Range`